src/plot_api/template_api.js

```diff
@@ -269,3 +269,187 @@ function getNextPath(parent, key, path) {
 
     return nextPath;
 }
+
+/**
+ * validateTemplate: Test for consistency between the given figure and
+ * a template, either already included in the figure or given separately.
+ * Note that not every issue we identify here is necessarily a problem,
+ * it depends on what you're using the template for.
+ *
+ * @param {object|DOM element} figure: the plot, with {data, layout} members,
+ *     to test the template against
+ * @param {Optional(object)} template: the template, with its own {data, layout},
+ *     to test. If omitted, we will look for a template already attached as the
+ *     plot's `layout.template` attribute.
+ *
+ * @returns {array} array of error objects each containing:
```

Comment:
THanks for making this have a similar API to `Plotly.validate` !

Reply:
One difference is I didn't make the object fields consistent - they all have `code` but each error, in fact even more specific than each `code`, includes whatever extra data it deems relevant. Given the nature of the issues (some apply to a specific trace, some to a path, some to a trace type but not a specific trace, etc... as opposed to `Plotly.validate` where you can basically always trace the issue back to a specific path) it seemed a bit to constraining to give them all the same fields.